synapse/storage/databases/main/roommember.py

```diff
@@ -1461,7 +1461,6 @@ def add_membership_profile_txn(txn: LoggingTransaction) -> int:
                 SELECT stream_ordering, event_id, events.room_id, event_json.json
                 FROM events
                 INNER JOIN event_json USING (event_id)
-                INNER JOIN room_memberships USING (event_id)
```

Comment:
I don't see any columns used from this table in the query

Reply:
This does make sure we only look at membership events but we also use `type = 'm.room.member'` below which does the same thing.